docs/predicate/slsa/v2.md

```diff
@@ -0,0 +1,621 @@
+        }
+      ]
+    }
+  }
+}
+```
+</details>
+<details>
+<summary><strong>TaskRun</strong></summary>
+    
+```json
+{
+  "_type": "https://in-toto.io/Statement/v0.1",
+  "predicateType": "https://slsa.dev/provenance/v1",
+  "subject": [
+    {
+      "name": "gcr.io/foo/bar",
+      "digest": {
+        "sha256": "05f95b26ed10668b7183c1e2da98610e91372fa9f510046d4ce5812addad86b5"
+      }
+    }
+  ],
+  "predicate": {
+    "buildDefinition": {
+      "buildType": "https://tekton.dev/predicate/chains/slsa/v2", 
+      "externalParameters": {
+        "runSpec": {
+          "serviceAccountName": "default",
+          "taskSpec": {
+            "steps": [
+              {
+                "name": "create-image",
+                "image": "busybox",
+                "resources": {},
+                "script": "#!/usr/bin/env sh\necho 'gcr.io/foo/bar' | tee $(results.IMAGE_URL.path)\necho 'sha256:05f95b26ed10668b7183c1e2da98610e91372fa9f510046d4ce5812addad86b5' | tee $(results.IMAGE_DIGEST.path)"
+              }
+            ],
+            "results": [
+              {
+                "name": "IMAGE_URL",
+                "type": "string"
+              },
+              {
+                "name": "IMAGE_DIGEST",
+                "type": "string"
+              }
+            ]
+          },
+          "timeout": "1h0m0s"
+        }
+      },
+      "internalParameters": {
+        "tekton-pipelines-feature-flags": {
+          "DisableAffinityAssistant": false,
+          "DisableCredsInit": false,
+          "RunningInEnvWithInjectedSidecars": true,
+          "RequireGitSSHSecretKnownHosts": false,
+          "EnableTektonOCIBundles": false,
+          "ScopeWhenExpressionsToTask": false,
+          "EnableAPIFields": "stable",
+          "SendCloudEventsForRuns": false,
+          "AwaitSidecarReadiness": true,
+          "EnforceNonfalsifiability": "",
+          "VerificationNoMatchPolicy": "ignore",
+          "EnableProvenanceInStatus": true,
+          "ResultExtractionMethod": "termination-message",
+          "MaxResultSize": 4096
+        }
+      },
+      "resolvedDependencies": [
+        {
+          "uri": "docker.io/library/busybox",
+          "digest": {
+            "sha256": "b5d6fe0712636ceb7430189de28819e195e8966372edfc2d9409d79402a0dc16"
+          },
+        }
+      ]
+    },
+    "runDetails": {
+      "builder": {
+        "id": "https://uri to your builder",
+        "builderDependencies": [
+             {
+                 "name": "chains",
+                 "uri": "git+https://github.com/tektoncd/chains.git",
+                 "digest": {"sha1": "4e11e2fe764ff8bddede42eee852767f7e5264c3"}
+             }
+        ],
+        "version": {
+              "tekton-pipelines": "v0.47.0",
+              "kubernetes": "1.27.0",
+              "platform": "arm64",
+              "os": "linux",
+        }
```

Comment:
IIUC, in the implementation https://github.com/tektoncd/chains/blob/70c8c7de563ba3d6c1e65a1e4e21c83335fa432a/pkg/chains/formats/slsa/v2alpha2/internal/taskrun/taskrun.go#L55-L57

we don't record anything other than builder id, but this example includes both version and builderDeps.

Reply:
Yes! I agree. Fixed.